src/store/user/userOperations.ts

```diff
@@ -0,0 +1,46 @@
+import { createAsyncThunk, nanoid } from "@reduxjs/toolkit";
+import { Token, User } from "../../types";
+import { setToken } from "../../axios";
+// import { axiosInstance } from "../../axios";
+
+export const login = createAsyncThunk<
+  { user: User; token: Token },
+  { email: string; password: string }
+>("user/login", async ({ email, password }, thunkAPI) => {
```

Comment:
We have the same types in userSlice. I thin better create types for 
```
{ user: User; token: Token },
{ email: string; password: string }
```
it will be more readable and reusable